src/schema/v2/artwork/collectorSignals.ts

```diff
@@ -179,6 +184,57 @@ export const CollectorSignals: GraphQLFieldConfig<any, ResolverContext> = {
   },
 }
 
+interface LabelSignalFields {
+  increasedInterest: boolean
+  // only a partial partner offer type
+  partnerOffer: { end_at: string; active: true } | null
+  primaryLabel: "PARTNER_OFFER" | "INCREASED_INTEREST" | null
+}
+
+// Single function to resolve mutually-exclusive label signals
+const getLabelSignalFields = async (
+  artwork,
+  ctx
+): Promise<LabelSignalFields> => {
+  const signals = {
+    increasedInterest: false,
+    partnerOffer: null,
+    primaryLabel: null,
+  } as LabelSignalFields
+
+  if (artwork.increased_interest_signal) {
+    signals.increasedInterest = true
+  }
+
+  if (
+    checkFeatureFlag("emerald_signals-partner-offers", ctx) &&
+    ctx.mePartnerOffersLoader &&
+    artwork.purchasable
+  ) {
+    const partnerOffers = await ctx.mePartnerOffersLoader({
+      artwork_id: artwork.id,
+      sort: "-created_at",
+      size: 1,
+    })
+
+    const activePartnerOffer = partnerOffers.body?.find((po) => po.active)
+
+    if (activePartnerOffer) {
+      signals.partnerOffer = activePartnerOffer
+    }
+  }
+
+  if (signals.partnerOffer) {
+    signals.primaryLabel = "PARTNER_OFFER"
```

Comment:
@damassi interested in your thoughts here: Should MP return copy?

My thinking is no because
- Tying copy changes to our MP schema will make updates more difficult in mobile
- returning a type (ie a string enum where each string is a value rather than copy) is more correct - `partner_offer` implies not just for label but also you will have a `collectorSignals.partnerOffer` to work with'. However as i mentioned in my PR description I didn't make this guarantee in the graphql types yet.

Reply:
We really don't want to encode UI logic into Metaphysics. What happens when you want to change it in only one place it's used, or only on Eigen, etc. If there's a constrained set of possible options then enum every time.